apps/crazyflie_position_publisher_reconnection/app.c

```diff
@@ -0,0 +1,187 @@
+#define RCSOFTCHECK(msg) if((rc != RCL_RET_OK)){DEBUG_PRINT("Failed status on line %d: %d. Continuing.\n",__LINE__,(int)rc);}
+
+static int pitchid, rollid, yawid;
+static int Xid, Yid, Zid;
+
+float sign(float x){   
+    return (x >= 0) ? 1.0 : -1.0;
+}
+
+void appMain(){ 
+    while(1){
+        absoluteUsedMemory = 0;
+        usedMemory = 0;
+
+        // ####################### RADIO INIT #######################
+
+        int radio_connected = logGetVarId("radio", "isConnected");
+
+        while(!logGetUint(radio_connected)) vTaskDelay(100);
+        DEBUG_PRINT("Radio connected\n");
+
+        // ####################### MICROROS INIT #######################
+
+        DEBUG_PRINT("Free heap pre uROS: %d bytes\n", xPortGetFreeHeapSize());
+        vTaskDelay(50);
+
+        rcl_context_t      context;
+        rcl_init_options_t init_options;
+        rcl_ret_t          rc;
+        rcl_ret_t          rc_aux __attribute__((unused));
+
+        init_options = rcl_get_zero_initialized_init_options();                          
+        rcl_allocator_t freeRTOS_allocator = rcutils_get_zero_initialized_allocator();   
+        freeRTOS_allocator.allocate = __crazyflie_allocate;
+        freeRTOS_allocator.deallocate = __crazyflie_deallocate;
+        freeRTOS_allocator.reallocate = __crazyflie_reallocate;
+        freeRTOS_allocator.zero_allocate = __crazyflie_zero_allocate;
+
+        if (!rcutils_set_default_allocator(&freeRTOS_allocator)) {                       
+            DEBUG_PRINT("Error on default allocators (line %d)\n",__LINE__); 
+            vTaskSuspend( NULL );
+        }
+
+        rc = rcl_init_options_init(&init_options, rcutils_get_default_allocator());      
+        RCCHECK()
+
+        context = rcl_get_zero_initialized_context();                                    
+
+        rc = rcl_init(0, NULL, &init_options, &context); 
+        RCCHECK()
+
+        rc = rcl_init_options_fini(&init_options);
+
+        rcl_node_t node = rcl_get_zero_initialized_node();
+        rcl_node_options_t node_ops = rcl_node_get_default_options();
+
+        rc = rcl_node_init(&node, "crazyflie_node", "", &context, &node_ops);
+        RCCHECK()
+
+        // Create publisher 1
+        const char* drone_odom = "/drone/odometry";
+
+        rcl_publisher_t pub_odom        = rcl_get_zero_initialized_publisher();
+        const rosidl_message_type_support_t * pub_type_support_odom = ROSIDL_GET_MSG_TYPE_SUPPORT(geometry_msgs, msg, Point32);
+        rcl_publisher_options_t pub_opt_odom = rcl_publisher_get_default_options();
+        pub_opt_odom.qos.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
+
+
+        rc = rcl_publisher_init(
+            &pub_odom,
+            &node,
+            pub_type_support_odom,
+            drone_odom,
+            &pub_opt_odom);
+
+        if(rc != RCL_RET_OK){
+            rc_aux = rcl_node_fini(&node);
```

Comment:
I think you could use
```suggestion
            (void) rcl_node_fini(&node);
```
instead of the `rc_aux` variable.

Reply:
This does not work with `-Werror` flag enabled in the CF Makefile when Release build:

```
error: ignoring return value of 'rcl_node_fini', declared with attribute warn_unused_result [-Werror=unused-result]
             (void) rcl_node_fini(&node);
```

Reply:
There is a wonderful [discussion](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425) about that in gcc.